File: solutions/java/CodeWars/src/main/java/com/kata/kyu6/stringSubpatternRecognitionI/main/StringSubpatternRecognitionI.java
package com.kata.kyu6.stringSubpatternRecognitionI.main;

public class StringSubpatternRecognitionI {
    public static boolean hasSubpattern(String str) {
        var duplicated = (str + str);
        duplicated = duplicated.substring(1, duplicated.length() - 1);
        return duplicated.contains(str);

    }
}

File: solutions/csharp/Kata/Kata/6-kyu/String_subpattern_recognition_I/src/StringSubpatternRecognitionI.cs
using System;

namespace Kata._6_kyu.String_subpattern_recognition_I.src;

public sealed class StringSubpatternRecognitionI
{
    public static bool HasSubpattern(string str) => 
        (str + str).IndexOf(str, 1, StringComparison.Ordinal) != str.Length;
    
}